iac-extensions/docker/src/test/java/org/sonar/iac/docker/visitors/DockerSymbolVisitorTest.java

```diff
@@ -0,0 +1,240 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.visitors;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import javax.annotation.Nullable;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.sonar.iac.common.api.tree.Tree;
+import org.sonar.iac.common.extension.visitors.InputFileContext;
+import org.sonar.iac.docker.parser.grammar.DockerLexicalGrammar;
+import org.sonar.iac.docker.symbols.Scope;
+import org.sonar.iac.docker.symbols.Symbol;
+import org.sonar.iac.docker.symbols.Usage;
+import org.sonar.iac.docker.tree.api.ArgInstruction;
+import org.sonar.iac.docker.tree.api.Body;
+import org.sonar.iac.docker.tree.api.DockerImage;
+import org.sonar.iac.docker.tree.api.HasScope;
+import org.sonar.iac.docker.tree.api.NewKeyValuePair;
+import org.sonar.iac.docker.tree.api.Variable;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.sonar.iac.docker.tree.impl.DockerTestUtils.parse;
+
+class DockerSymbolVisitorTest {
+
+  private static final String IMAGE_PREFIX = "FROM foobar\n";
+  private final InputFileContext inputFileContext = mock(InputFileContext.class);
+
+  @Test
+  void argInstructionShouldCreateSymbol() {
+    DockerImage image = scanImage("ARG foo=bar");
+    NewKeyValuePair keyValuePair = firstDescendant(image, NewKeyValuePair.class);
+
+    Scope scope = image.scope();
+
+    assertThat(scope.getSymbols()).hasSize(1);
+    assertThat(image.scope().getSymbol("bar")).isNull();
+
+    Symbol symbol = image.scope().getSymbol("foo");
+    assertThat(symbol).isNotNull();
+    assertThat(symbol.usages()).allSatisfy(usage -> {
```

Comment:
I would add an assert to ensure that there is only one usage, just in case, but maybe it's too much ^^'